src/transaction/log_page_buffer.c

```diff
@@ -412,58 +372,40 @@ static void logpb_append_crumbs (THREAD_ENTRY * thread_p, int num_crumbs, const
 static void logpb_next_append_page (THREAD_ENTRY * thread_p, LOG_SETDIRTY current_setdirty);
 static LOG_PRIOR_NODE *prior_lsa_remove_prior_list (THREAD_ENTRY * thread_p);
 static int logpb_append_prior_lsa_list (THREAD_ENTRY * thread_p, LOG_PRIOR_NODE * list);
-static LOG_PAGE *logpb_copy_page (THREAD_ENTRY * thread_p, LOG_PAGEID pageid, LOG_CS_ACCESS_MODE access_mode,
-				  LOG_PAGE * log_pgptr);
+static int logpb_copy_page (THREAD_ENTRY * thread_p, LOG_PAGEID pageid, LOG_CS_ACCESS_MODE access_mode,
+			    LOG_PAGE * log_pgptr);
 
 static void logpb_fatal_error_internal (THREAD_ENTRY * thread_p, bool log_exit, bool need_flush, const char *file_name,
 					const int lineno, const char *fmt, va_list ap);
 
 static void logpb_set_nxio_lsa (LOG_LSA * lsa);
 
 static int logpb_copy_log_header (THREAD_ENTRY * thread_p, LOG_HEADER * to_hdr, const LOG_HEADER * from_hdr);
-
+STATIC_INLINE LOG_BUFFER *logpb_get_log_buffer (LOG_PAGE * log_pg) __attribute__ ((ALWAYS_INLINE));
+STATIC_INLINE int log_get_log_buffer_index (LOG_PAGEID log_pageid);
 /*
  * FUNCTIONS RELATED TO LOG BUFFERING
  *
  */
 
-/*
- * logpb_reset_clock_hand - reset clock hand of log page buffer
- *
- * return: nothing
- *
- * NOTE:
- *
- */
-static void
-logpb_reset_clock_hand (int buffer_index)
+STATIC_INLINE int
+log_get_log_buffer_index (LOG_PAGEID log_pageid)
 {
-  if (buffer_index >= log_Pb.num_buffers || buffer_index < 0)
-    {
-      log_Pb.clock_hand = 0;
-    }
-  else
-    {
-      log_Pb.clock_hand = buffer_index;
-    }
+  return log_pageid % log_Pb.num_buffers;
 }
 
-/*
- * logpb_move_next_clock_hand - move next clock hand of log page buffer
- *
- * return: nothing
- *
- * NOTE:
- *
- */
-static void
-logpb_move_next_clock_hand (void)
+LOG_BUFFER *
+logpb_get_log_buffer (LOG_PAGE * log_pg)
 {
-  log_Pb.clock_hand++;
-  if (log_Pb.clock_hand >= log_Pb.num_buffers)
-    {
-      log_Pb.clock_hand = 0;
-    }
+
+  long long index = ((char *) log_pg - (char *) log_Pb.pages_area) / LOG_PAGESIZE;
+  if (log_pg == log_Pb.header_page)
+    return &log_Pb.header_buffer;
+  /* Safe guard: index is valid. */
+  assert (index >= 0 && index < log_Pb.num_buffers);
+  /* Safe guard: log_pg is correctly aligned. */
+  assert ((char *) log_Pb.pages_area + LOG_PAGESIZE * index == (char *) log_pg);
+  return &log_Pb.buffers[index];
```

Comment:
@funkydvd 

I'd like to propose 

```C
  if (log_pg == log_Pb.header_page)
    {
      return &log_Pb.header_buffer;
    }

  /* Safe guard: index is valid. */
  assert (index >= 0 && index < log_Pb.num_buffers);

  /* Safe guard: log_pg is correctly aligned. */
  assert ((char *) log_Pb.pages_area + (LOG_PAGESIZE * index) == (char *) log_pg);

  return &log_Pb.buffers[index];
```

Please let me remind you our coding convention. 
We always use explicit block even though it is a single statement. It prevents developers from committing unintended misses. 

Please also don't spare horizontal blank lines. It will greatly help developers to easily read codes.